theano/tensor/opt.py

```diff
@@ -3888,6 +3888,41 @@ def local_sum_mul_by_scalar(node):
         if thing_summed.owner and thing_summed.owner.op == T.neg:
             return [T.neg(node.op(thing_summed.owner.inputs[0]))]
 
+@register_specialize
+@gof.local_optimizer([T.elemwise.Prod])
+def local_prod_mul_by_scalar(node):
+    """
+    Convert a prod(scalar * smth) -> scalar * prod(smth)
+
+
+    Really, the logical of this opt is exactly the same as local_sum_mul_by_scalar,
+    and this is just code duplication
```

Comment:
Then why are you copying it?  It would be better to integrate the small changes in the opt above than to copy all the code.